Evaluation/apply_training.py

```diff
@@ -0,0 +1,107 @@
+import os
+import sys
+import json
+import git
+from tqdm import tqdm
+
+import uproot
+import numpy as np
+import pandas as pd
+from tensorflow.keras.models import load_model
+
+import mlflow
+import hydra
+from hydra.utils import to_absolute_path
+from omegaconf import DictConfig, OmegaConf
+
+sys.path.insert(0, "../Training/python")
+from common import setup_gpu
+
+@hydra.main(config_path='.', config_name='apply_training')
+def main(cfg: DictConfig) -> None:
+    # set up paths & gpu
+    mlflow.set_tracking_uri(f"file://{to_absolute_path(cfg.path_to_mlflow)}")
+    path_to_artifacts = to_absolute_path(f'{cfg.path_to_mlflow}/{cfg.experiment_id}/{cfg.run_id}/artifacts/')
+    setup_gpu(cfg.gpu_cfg)
+
+    # load the model
+    with open(to_absolute_path(f'{path_to_artifacts}/input_cfg/metric_names.json')) as f:
+        metric_names = json.load(f)
+    path_to_model = f'{path_to_artifacts}/model'
+    model = load_model(path_to_model, {name: lambda _: None for name in metric_names.keys()}) # workaround to load the model without loading metric functions
+
+    # load baseline training cfg and update it with parsed arguments
+    training_cfg = OmegaConf.load(to_absolute_path(cfg.path_to_training_cfg))
+    if cfg.training_cfg_upd is not None:
+        training_cfg = OmegaConf.merge(training_cfg, cfg.training_cfg_upd)
+    training_cfg = OmegaConf.to_object(training_cfg)
+
+    # fetch historic git commit used to run training 
+    with mlflow.start_run(experiment_id=cfg.experiment_id, run_id=cfg.run_id) as active_run:
+        train_git_commit = active_run.data.params['git_commit'] if 'git_commit' in active_run.data.params else None
```

Comment:
- Please, add the possibility to disable the checkout step (keeping the current behaviour by default). In some cases, one could want to have a modified DataLoader. 
- if `params` is `dict`-like use `get` instead

Reply:
Sure, added `checkout_train_repo` switch in 06ae65f. As for `params`, this is an attribute of the mlflow class, so can't change it to a dict-like interface here

Reply:
sorry, is it possible or not to use `get`? Because in the current version of the code it is changed to `get`, but based on your reply it shouldn't be done.

Reply:
Well, since it is a nested call, it is not possible for the first steps ([`data`](https://www.mlflow.org/docs/latest/python_api/mlflow.entities.html#mlflow.entities.Run.data) and [`params`](https://www.mlflow.org/docs/latest/python_api/mlflow.entities.html#mlflow.entities.RunData.params) are the class properties) but possible for `git_commit` (key in the dictionary) 🙂 I changed the behaviour for `git_commit` to `get()` because in that case it will directly return `None` if this key doesn't exist. So I don't have to explicitly check this with `if 'git_commit' in active_run.data.params` and it looks more compact